Refactor/Add reviewerName field to review response

yeong-hwan opened the pull request:
### PR 요약

리뷰 응답에 `reviewerName` 필드 추가 및 `findByPortfolioId` 메소드 리팩토링

### 변경 사항
- PortfolioService의 mapToReviewResponse 메소드에 reviewerName 필드를 추가하여 리뷰어의 이름을 응답 DTO에 포함시킴.
- PortfolioService에서 Portfolio 객체를 ID로 조회한 후 리뷰를 가져오도록 수정.
- ReviewRepository의 findByPortfolioId 메소드를 findByPortfolio로 리팩토링하여 portfolioId 대신 Portfolio 객체를 인수로 받도록 변경.
- PortfolioService의 import 구문을 변경된 로직에 맞게 조정.
- PortfolioService의 일부 포맷팅 문제를 수정.

### 참고 사항

yeong-hwan committed:
refactor: modify `findByPortfolioId` method to use `Portfolio` object

- Updated `PortfolioService` to retrieve `Portfolio` by ID before fetching reviews.
- Refactored `ReviewRepository` method `findByPortfolioId` to `findByPortfolio` to accept a `Portfolio` object instead of `portfolioId`.
- Adjusted the import statements to accommodate changes in `PortfolioService`.
- Fixed minor formatting issues in `PortfolioService`.

yeong-hwan committed:
refactor: add `reviewerName` field to review response mapping

- Added the `reviewerName` field to the `mapToReviewResponse` method in `PortfolioService` to include the reviewer's name in the response DTO.

yeong-hwan committed:
refactor: remove presignedWeddingPhotoUrls handling from ReviewService

- Removed all instances of `presignedWeddingPhotoUrls` from the `ReviewService`, `ReviewDTO`, and related files.
- Added `weddingPhotoUrls` handling in `PortfolioService` and `ReviewDTO`.
- Improved code formatting and corrected a minor comment typo in `ReviewService`.
- Removed unnecessary imports in `DataLoader`.

yeong-hwan commented on demo/src/main/java/com/example/demo/portfolio/service/PortfolioService.java:
```diff
@@ -274,13 +276,15 @@ private void updatePortfolioWithModifiedReview(Portfolio portfolio, ReviewDTO.Re
                 .radar(review.getRadar())
                 .content(review.getContent())
                 .createdAt(review.getCreatedAt())
                 .updatedAt(review.getUpdatedAt())
```
해당 부분 `photoUrls` 추가하여 리팩토링 했습니다

yeong-hwan commented on demo/src/main/java/com/example/demo/portfolio/service/PortfolioService.java:
```diff
@@ -274,13 +276,15 @@ private void updatePortfolioWithModifiedReview(Portfolio portfolio, ReviewDTO.Re
                 .radar(review.getRadar())
                 .content(review.getContent())
                 .createdAt(review.getCreatedAt())
                 .updatedAt(review.getUpdatedAt())
+                .weddingPhotoUrls(review.getWeddingPhotoUrls())
                 .build();
```
해당 부분 `photoUrls` 추가하여 리팩토링 했습니다

yeong-hwan commented:
response에 `presignedUrls`는 필요없어서 전부 삭제했습니다.

enihsgnir reviewed (changes requested):
포트폴리오 DTO도 마찬가지이지만 단순히 조회할 땐 presigned url이 필요없지만, POST /api/v1/review/customer/create 처럼 이미지가 포함된 데이터를 새로 생성하는 API 요청에 대한 응답 바디에는 presigned url이 있어야 이미지를 클라이언트가 직접 업로드 할 수 있습니다!

조회용 DTO와 생성에 대한 응답 DTO가 구분되어야 할 것 같습니다.

yeong-hwan commented:
> 포트폴리오 DTO도 마찬가지이지만 단순히 조회할 땐 presigned url이 필요없지만, POST /api/v1/review/customer/create 처럼 이미지가 포함된 데이터를 새로 생성하는 API 요청에 대한 응답 바디에는 presigned url이 있어야 이미지를 클라이언트가 직접 업로드 할 수 있습니다!
> 
> 
> 
> 조회용 DTO와 생성에 대한 응답 DTO가 구분되어야 할 것 같습니다.

로직을 자주 까먹네요 ㅠ 해당 부분 롤백 해두겠습니다!

yeong-hwan committed:
refacotr: rollback deletion of presignedUrl list

yeong-hwan committed:
refactor: change native SQL queries to JPQL in ReviewRepository

- Changed native SQL queries to JPQL queries in `ReviewRepository` for better integration with JPA.
- Updated queries for soft-deleted reviews, reviews for wedding planners, and reviews for customers.
- Improved readability and maintainability of query methods.
